### PR TITLE
Notify Admin if stolen bike with promoted alert is recovered

### PR DESCRIPTION
**Pull request by jmromer**
When marking a bike as recovered, send an admin notification if the bike has an active promoted alert.

Screen Shot 2019-11-07 at 12 46 23 PM

Small UI fix (render the CTA as a button rather than a link):

Screen Shot 2019-11-07 at 12 40 28 PM

### CODE REVIEW

File: app/controllers/bikes/recovery_controller.rb [hunk 1]
```diff
@@ -16,7 +16,8 @@ def update
         flash[:success] = translation(:bike_recovered)
         redirect_to bike_path(@bike)
       else
-        render :edit, bike_id: @bike.id, token: params[:token]
+        session[:recovery_link_token] = params[:token]
```
Comment (sethherr) [inline app/controllers/bikes/recovery_controller.rb]: Hot take: SICK! Good addition, makes life better for everyone.

File: app/mailers/admin_mailer.rb [hunk 2]
```diff
@@ -41,11 +41,20 @@ def lightspeed_notification_email(organization, api_key)
...
 
-  def theft_alert_purchased(theft_alert)
+  def theft_alert_purchased(theft_alert, notify_of_recovery: false)
     @theft_alert = theft_alert
     @theft_alert_plan = theft_alert.theft_alert_plan
     @creator = theft_alert.creator
     @bike = theft_alert.bike
-    mail(to: "stolenbikealerts@bikeindex.org", subject: "Theft Alert purchased: #{@theft_alert.id}")
+
+    if notify_of_recovery
```
Comment (sethherr) [inline app/mailers/admin_mailer.rb]: This confused me at first, that both emails where in the same mailer method

Comment (sethherr) [reply app/mailers/admin_mailer.rb]: I would like it to be made clearer, somehow?

I think being in the same method makes sense. Idea for how to resolve:  mailer method was renamed `theft_alert_notification`, second arg was `notification_type: "purchased"`

My

File: app/controllers/bikes/recovery_controller.rb (hunk 1 above)
Comment (sethherr) [reply app/controllers/bikes/recovery_controller.rb]: Oh. This was something that should've been there already. Oops, my bad. Any way you can add a test?

File: app/mailers/admin_mailer.rb (hunk 2 above)
Comment (jmromer) [reply app/mailers/admin_mailer.rb]: Good call thanks — updated in 1998c27d

**Pull request merged**